### DISCUSSION TITLE
[Question] How to use FlowSettings

### DISCUSSION THREAD

**Discussion by fodeguirassy** [russhwolf/multiplatform-settings#110] [Q&A]
Hello

Thanks for this great library you're shipping to us.

I was looking for a basic usage of FlowSettings.  I tried the following but getting errors.

![Screenshot 2021-07-20 at 15 01 25](https://user-images.githubusercontent.com/16901741/126329017-1e86799c-2ce4-4b13-8cd3-c8c6056c87a5.png)

Thanks for your help !

**Comment by russhwolf** [comment]
What are the errors you're getting?

Try
```
val flowSettings: FlowSettings = sharedSettings.toFlowSettings()
val flow: Flow<Boolean> = flowSettings.getBooleanFlow("key", false)
```

**Comment by fodeguirassy** [comment]
`toFlowSettings()` seems unavailable.
![Screenshot 2021-07-21 at 20 47 50](https://user-images.githubusercontent.com/16901741/126551351-5bc616a9-2be0-402b-8070-7aa544af7e38.png)

**Comment by russhwolf** [comment]
Oh sorry. `sharedSettings` needs to be `ObservableSettings` to be able to convert to flow.

**Reply by EGAMAGZ** [reply]
Please can you give an example of how to use ObservableSettings, at least I still don't understand how to use it?

**Comment by chris-hatton** [comment]
I'm also stuck trying to use the Flows API's - it looks like they are either missing or out of sync with the documentation.  I am on Desktop JVM.

**Comment by russhwolf** [comment]
Can you say more about what you're trying to do? I believe the documentation is correct and up-to-date but maybe there are missing details that are causing confusion.

Do be aware that `Flow` APIs are only available for `ObservableSettings` instances, rather than for all `Settings`. If you're using the `no-arg` module you'll need to cast the instance returned by `Settings()` (but be aware that this won't work on all platforms).
